apps/plausiblestats/plausible_stats.star

```diff
@@ -0,0 +1,569 @@
+def decorate_value(value, decimal_index, suffix):
+    # Convert the string to a list
+    value_list = list(value.elems())
+
+    # Take the first 4 characters
+    cropped_list = value_list[:4]
+
+    # Insert the "." character at the decimal_index
+    cropped_list.insert(decimal_index, ".")
+
+    # Smash it back into a string
+    joined = "".join(cropped_list)
+
+    # Loop through and remove any and all trailing "0" characters
+    for i in range(len(joined)):
+        joined = joined.removesuffix("0")
+
+    # Remove a trailing decimal separator if present
+    joined = joined.removesuffix(".")
+
+    # Return the joined string, with the suffix added
+    return joined + suffix
+
+# Removes the chance for human error. Does its best to remove
+# the scheme and "www" subdomain if present.
+def sanitize_domain(domain):
+    # Check for empty at first
+    if domain == "" or domain == None:
+        return None
+
+    # Remove whitespace
+    stripped_domain = domain.lstrip()
+
+    # Strip out "http://" or "https://"
+    prefix_free_domain = stripped_domain.split("://").pop()
+
+    # Remove "www."
+    final_url = prefix_free_domain.removeprefix("www.")
+
+    # Don't bother
+
+    return final_url
+
+# Makes a request to the domain, and will attempt to return the site's favicon
+# icon by assuming the three most common favicon filenames.
+# Defaults to GLOBE_IMAGE if it fails.
+def get_favicon(domain, favicon_path):
+    cache_id = FAVICON_CACHE_ID + "_" + domain
+
+    cached_favicon = cache.get(cache_id)
+
+    if cached_favicon != None:
+        print("Returning cached favicon")
+        return cached_favicon
+
+    base_url = "http://" + domain
+
+    if favicon_path != None:
+        print("Using custom favicon path")
+        favicon_url = base_url + favicon_path
+        response = http.get(favicon_url)
+        if response.status_code != 200:
+            print("Failed to get favicon: %d" % response.status_code)
+            return None
+        favicon = response.body()
+        cache.set(cache_id, favicon, ttl_seconds = FAVICON_CACHE_TTL)
+        print("Got custom favicon")
+        return favicon
+
+    for f in FAVICON_FILENAMES:
+        response = http.get(base_url + f)
+        if response.status_code != 200:
+            continue
+        favicon = response.body()
+        cache.set(cache_id, favicon, ttl_seconds = FAVICON_CACHE_TTL)
+        return favicon
+
+    return GLOBE_IMAGE
+
+# Makes a call to the plausible.io stats endpoint.
+#    endpoint: the path to the API to call eg. "/aggregate" or "/timeseries"
+#    toke: The Auth token
+#    domain: The domain to check
+#    time_period: The time period to check (doesn't support custom date ranges)
+#    metric: The metric value to return
+def get_plausible_data(endpoint, token, domain, time_period, metric):
+    print("Getting data from Plausible API")
+
+    cache_id = "_".join([REQUEST_CACHE_ID, endpoint, domain, time_period, metric])
```

Comment:
I think you need `token` to be part of the cache key. Otherwise any user will be able to call up cache records for any domain for which another user has provided the correct token.

Reply:
That's a really good point. I think my mental model of the cache didn't include the idea that it was shared across everyone. But that makes sense. I've made that change to include it.